data inconsistent in chaos test

### What did you do?

run chaos test (pod-kill-dataflow)

ref: https://github.com/pingcap/tiflow/actions/runs/3292268443/jobs/5427428684

detail logs: 
[chaos-base-logs.pod-kill-dataflow.zip](https://github.com/pingcap/tiflow/files/9836365/chaos-base-logs.pod-kill-dataflow.zip)


### What did you expect to see?

_No response_

### What did you see instead?

```go
[2022/10/20 19:27:46.175 +00:00] [INFO] [checkpoint.go:276] ["summary info"] [instance_id=target] [schema=dmjob] [table=tb1] ["chunk num"=3] ["success num"=2] ["failed num"=1] ["ignore num"=0]
[2022/10/20 19:27:46.179 +00:00] [INFO] [case.go:152] ["finish run case"] [name=dmjob] [job_id=ecd924da-10db-4a49-a1a2-ad2e098698cc] [insert=10861] [update=10545] [delete=10694]
{"level":"warn","ts":"2022-10-20T19:28:06.980Z","logger":"etcd-client","caller":"v3@v3.5.2/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000f72700/chaos-metastore-etcd:12479","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}
Error: 0/20 19:28:06.986 +00:00] [ERROR] [main.go:87] ["run cases failed"] [error="different data for table `dmjob`.`tb1`"] [errorVerbose="different data for table `dmjob`.`tb1`\ngithub.com/pingcap/tiflow/engine/chaos/cases/dm.syncDiffInspector\n\tgithub.com/pingcap/tiflow/engine/chaos/cases/dm/case.go:408\ngithub.com/pingcap/tiflow/engine/chaos/cases/dm.(*Case).diffDataLoop\n\tgithub.com/pingcap/tiflow/engine/chaos/cases/dm/case.go:263\ngithub.com/pingcap/tiflow/engine/chaos/cases/dm.(*Case).incrLoop\n\tgithub.com/pingcap/tiflow/engine/chaos/cases/dm/case.go:347\ngithub.com/pingcap/tiflow/engine/chaos/cases/dm.(*Case).Run\n\tgithub.com/pingcap/tiflow/engine/chaos/cases/dm/case.go:164\nmain.runDMJobCases.func1\n\tgithub.com/pingcap/tiflow/engine/chaos/cases/case_dm_job.go:36\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:75\nruntime.goexit\n\truntime/asm_amd64.s:1594"]
```

### Versions of the cluster

Dataflow Engine version (run `tiflow version`):

master@https://github.com/pingcap/tiflow/commit/4fa939d91515a87bbe6ce8b3fc9559d18c4d3798

--- a/deployments/engine/helm/minio/minio.yaml
+++ b/deployments/engine/helm/minio/minio.yaml
@@ -67,7 +67,6 @@ spec:
     spec:
       accessModes:
         - ReadWriteOnce
-      storageClassName: standard
       resources:
         requests:
           storage: 10Gi

--- a/engine/chaos/cases/dm/case.go
+++ b/engine/chaos/cases/dm/case.go
@@ -264,7 +264,7 @@ func (c *Case) diffDataLoop(ctx context.Context) error {
 }
 
 // randDML generates DML (INSERT, UPDATE or DELETE).
-func (c *Case) randDML(source int, table string) (string, error) {
+func (c *Case) randDML(source int, table string, deleteKeys map[string][]string) (string, error) {
 	generator := c.generators[source][table]
 	mcp := c.mcps[source][table]
 	t := rand.Intn(3)
@@ -300,7 +300,7 @@ func (c *Case) randDML(source int, table string) (string, error) {
 		if err != nil {
 			return "", err
 		}
-		delete(c.keySet[table], key.GetValueHash())
+		deleteKeys[table] = append(deleteKeys[table], key.GetValueHash())
 		err = mcp.DeleteUK(key)
 		return sql, err
 	}
@@ -308,6 +308,14 @@ func (c *Case) randDML(source int, table string) (string, error) {
 
 func (c *Case) genIncrData(ctx context.Context) error {
 	log.L().Info("start generate incremental data", zap.String("name", c.name), zap.String("job_id", c.jobID))
+	deleteKeys := make(map[string][]string)
+	defer func() {
+		for tb, keys := range deleteKeys {
+			for _, k := range keys {
+				delete(c.keySet[tb], k)
+			}
+		}
+	}()
 	for {
 		select {
 		case <-ctx.Done():
@@ -319,7 +327,7 @@ func (c *Case) genIncrData(ctx context.Context) error {
 
 		sqls := make([]string, 0, batch)
 		for i := 0; i < batch; i++ {
-			sql, err := c.randDML(source, tableName)
+			sql, err := c.randDML(source, tableName, deleteKeys)
 			if err != nil {
 				return err
 			}

